wgpu-core/src/device/life.rs

```diff
@@ -419,10 +419,14 @@ impl<B: GfxBackend> LifetimeTracker<B> {
             .keys()
             .filter_map(|key| {
                 let mut last_submit: SubmissionIndex = 0;
+                let mut has_missing_view = false;
+
                 for &at in key.all() {
+                    // Clean the framebuffer up if at least one view it's referring to is gone
                     if texture_view_guard.contains(at) {
-                        return None;
+                        has_missing_view = true;
```

Comment:
looks like the logic is all messed up here.

First of all, ` if texture_view_guard.contains(at)` means that the view is NOT gone, the opposite from the comment and the logic inside.

Secondly, the follow-up code says "This attachment is no longer registered.", it's only meant to run if the view is missing.

Reply:
Oh! My bad yea you're completely right, I'll take another look over it.